packages/core/src/render3/i18n_debug.ts

```diff
@@ -0,0 +1,186 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {assertNumber, assertString} from '../util/assert';
+
+import {COMMENT_MARKER, ELEMENT_MARKER, getInstructionFromI18nMutateOpCode, getParentFromI18nMutateOpCode, getRefFromI18nMutateOpCode, I18nMutateOpCode, I18nMutateOpCodes, I18nUpdateOpCode, I18nUpdateOpCodes} from './interfaces/i18n';
+
+/**
+ * Converts `I18nUpdateOpCodes` array into a human readable format.
+ *
+ * This function is attached to the `I18nUpdateOpCodes.debug` property if `ngDevMode` is enabled.
+ * This function provides a human readable view of the opcodes. This is useful when debugging the
+ * application as well as writing more readable tests.
+ *
+ * @param this `I18nUpdateOpCodes` if attached as a method.
+ * @param opcodes `I18nUpdateOpCodes` if invoked as a function.
+ */
+export function i18nUpdateOpCodesToString(
+    this: I18nUpdateOpCodes|void, opcodes?: I18nUpdateOpCodes): string[] {
+  const parser = new OpCodeParser(opcodes || (Array.isArray(this) ? this : []));
+  let lines: string[] = [];
+
+  function consumeOpCode(value: number): string {
+    const ref = value >>> I18nUpdateOpCode.SHIFT_REF;
+    const opCode = value & I18nUpdateOpCode.MASK_OPCODE;
+    switch (opCode) {
+      case I18nUpdateOpCode.Text:
+        return `(lView[${ref}] as Text).textContent = $$$`;
+      case I18nUpdateOpCode.Attr:
+        const attrName = parser.consumeString();
+        const sanitizationFn = parser.consumeFunction();
+        const value = sanitizationFn ? `(${sanitizationFn})($$$)` : '$$$';
+        return `(lView[${ref}] as Element).setAttribute('${attrName}', ${value})`;
+      case I18nUpdateOpCode.IcuSwitch:
+        return `icuSwitchCase(lView[${ref}] as Comment, ${parser.consumeNumber()}, $$$)`;
+      case I18nUpdateOpCode.IcuUpdate:
+        return `icuUpdateCase(lView[${ref}] as Comment, ${parser.consumeNumber()})`;
+    }
+    throw 'unexpected OpCode';
```

Comment:
```suggestion
    throw new Error('unexpected OpCode');
```
(to be consistent with the rest of the code)

Reply:
fixed